runner/util_linux.go

```diff
@@ -4,19 +4,30 @@ import (
 	"io"
 	"os/exec"
 	"syscall"
+	"time"
 
 	"github.com/creack/pty"
 )
 
-func killCmd(cmd *exec.Cmd) (int, error) {
-	pid := cmd.Process.Pid
+func (e *Engine) killCmd(cmd *exec.Cmd) (pid int, err error) {
+	pid = cmd.Process.Pid
+
+	if e.config.Build.SendInterrupt {
+		// Sending a signal to make it clear to the process that it is time to turn off
+		if err = syscall.Kill(-pid, syscall.SIGINT); err != nil {
+			return
+		}
+
+		time.Sleep(e.config.Build.KillDelay * time.Second)
+	}
+
 	// https://stackoverflow.com/questions/22470193/why-wont-go-kill-a-child-process-correctly
-	err := syscall.Kill(-pid, syscall.SIGKILL)
+	err = syscall.Kill(-pid, syscall.SIGKILL)
 
 	// Wait releases any resources associated with the Process.
 	_, _ = cmd.Process.Wait()
 
-	return pid, err
+	return
```

Comment:
As per golang docs:
> // The only signal values guaranteed to be present in the os package on all
> // systems are os.Interrupt (send the process an interrupt) and os.Kill (force
> // the process to exit). On Windows, sending os.Interrupt to a process with
> // os.Process.Signal is not implemented; it will return an error instead of
> // sending a signal.
So I do not add `syscall.SIGINT` part to `windows`. Also I do not add `cmd.Process.Wait` , `kill.Run()` have it already.
To `darwin` I added all.
I do not have windows or macos, so I can not test these changes.

Reply:
I see. I'll test macOS.